### PR TITLE
Update OperandRegistry and OperandConfig for SaaS deployment

### PR DESCRIPTION
**Pull request by Daniel-Fan**
What is this PR doing?

1. In CSV, Create brand new templates of `OperandConfig` and `OperandRegistry` for SaaS deployment.
2. Integrate `ControlNs` and `SaasEnable` into `bootstrap`
3. Integrate the installation flow of Namespace-scope operator into one function in order to reduce the complexity of `InitResources` function.
4. don't install `secretshare operator` in SaaS deployment https://github.com/IBM/ibm-common-service-operator/pull/446#discussion_r605361604
5. Install different `OperandConfig` and `OperandRegistry` based on SaaS or non-SaaS deployment. https://github.com/IBM/ibm-common-service-operator/pull/450#issuecomment-809105028

What is unclear?
1. the content of templates for both `OperandConfig` and `OperandRegistry`

### CODE REVIEW

**Comment by horis233** [comment]
@Daniel-Fan 
I am wondering if we can manage another OperandRegistry template for SaaS since not all the services will be enabled in SaaS deployment.

cc @bitscuit

**Comment by Daniel-Fan** [comment]
@horis233 

Yes, of course.

But no matter which method we choose, patching a new Registry to the original one or having a brand new registry template for SaaS deployment, We both need to `namespacelize` the Registry twice.
Or
We consider to have two different registries concurrently, one for services installed in shared namespace and another one for the services installed in masternamespce.

**Comment by horis233** [comment]
> We consider to have two different registries concurrently, one for services installed in shared namespace and another one for the services installed in masternamespce.

Please don't create two different registries, which is hard to maintain.

> We both need to namespacelize the Registry twice.

Do you mean namespacelize once for the cluster singleton, once for those services? The essence of namespacelize is replacing placeholders in the string. I believe we can use different placeholders for controller namespace and cs master namespace.

**Comment by Daniel-Fan** [comment]
> Please don't create two different registries, which is hard to maintain.
Yes, I agree with it.

> Do you mean namespacelize once for the cluster singleton, once for those services? The essence of namespacelize is replacing placeholders in the string. I believe we can use different placeholders for controller namespace and cs master namespace.
Yes, if we choose to have two separate `OperandRegistry` for SaaS deployment and non-SaaS deployment. Then using different `placeholder` for twice `namespacelize` would be a easy and simply way to deploy `OperandRegistry`.

Also we need to nail down the `OperandRegistry` template for SaaS deployment as well.

**Comment by Daniel-Fan** [comment]
@horis233 @bitscuit 

This PR is ready for reviewing, please see the details of this PR here https://github.com/IBM/ibm-common-service-operator/pull/450#issue-602027752

File: bundle/manifests/ibm-common-service-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -244,6 +321,134 @@ metadata:
...
+          scope: public
+          sourceName: opencloud-operators
+          sourceNamespace: openshift-marketplace
+        - channel: v1.0
+          name: ibm-db2u-operator
+          namespace: placeholder
+          packageName: db2u-operator
+          scope: public
+          sourceName: ibm-operator-catalog
+          sourceNamespace: openshift-marketplace
```
Comment (horis233) [inline bundle/manifests/ibm-common-service-operator.clusterserviceversion.yaml:451]: @Daniel-Fan  let's remove `ibm-db2u-operator`, `redhat-marketplace-operator`, `ibm-monitoring-grafana-operator`, `ibm-monitoring-prometheusext-operator`, `ibm-monitoring-exporters-operator` `ibm-commonui-operator` and `ibm-healthcheck-operator` from the OperandRegistry

Comment (Daniel-Fan) [reply bundle/manifests/ibm-common-service-operator.clusterserviceversion.yaml]: Sure, updated their `OperandConfig` and `OperandRegistry`

**Review by horis233 — APPROVED** [review]
/lgtm

**Comment by horis233** [comment]
@Daniel-Fan Could you help to rebase this PR?

**Review by horis233 — APPROVED** [review]
/lgtm

**Pull request merged**